### PR TITLE
Add size based assertions

### PR DESCRIPTION
**Pull request by tttc3**
Adds size based assertions: `assert_size`, `assert_equal_size`, `assert_trees_all_equal_sizes`. The additions are based on the `shape` based equivalents. 

These assertions are useful for one of my own use cases, so thought they might be a nice addition to the package.

### CODE REVIEW

File: chex/__init__.py [hunk 1]
```diff
@@ -58,6 +60,7 @@
 from chex._src.asserts import assert_trees_all_equal
 from chex._src.asserts import assert_trees_all_equal_comparator
 from chex._src.asserts import assert_trees_all_equal_dtypes
+from chex._src.asserts import assert_trees_all_equal_sizes
```
Comment (hbq1) [inline chex/__init__.py:63]: Missing this one in \_\_all\_\_.

**Review by hbq1 — APPROVED** [review]
This is awesome! Thanks a lot for the PR!

File: chex/__init__.py (hunk 1 above)
Comment (tttc3) [reply chex/__init__.py:63]: Good spot, this is sorted in the latest commit.

**Review by hbq1 — APPROVED** [review]
Thanks!

**Pull request merged**